provisioning/transport/mqtt/src/mqtt.ts

```diff
@@ -52,11 +53,12 @@ export class Mqtt extends EventEmitter implements X509ProvisioningTransport, Sym
 
       /* Codes_SRS_NODE_PROVISIONING_MQTT_18_010: [ When waiting for responses, `registrationRequest` shall watch for messages with a topic named $dps/registrations/res/<status>/?$rid=<rid>.] */
       /* Codes_SRS_NODE_PROVISIONING_MQTT_18_024: [ When waiting for responses, `queryOperationStatus` shall watch for messages with a topic named $dps/registrations/res/<status>/?$rid=<rid>.] */
-      let match = topic.match(/^\$dps\/registrations\/res\/(.*)\/\?\$rid=(.*)$/);
+      let match = topic.match(/^\$dps\/registrations\/res\/(.*)\/\?(.*)$/);
+      let queryParameters = queryString.parse(match[2]);
 
-      if (!!match && match.length === 3) {
+      if (!!match && match.length === 3 && queryParameters.$rid) {
         let status: number = Number(match[1]);
-        let rid: string = match[2];
+        let rid: string = queryParameters.$rid.toString();
```

Comment:
>.toString() [](start = 46, length = 11)

nit: given the querystring is key/value pairs of strings already, i think this is unnecessary?

Reply:
Will cast to string.

---
In reply to: [280186974](https://github.com/Azure/azure-iot-sdk-node/pull/549#discussion_r280186974) [](ancestors = 280186974)